### PR TITLE
Upgrade tonic + cors layer

### LINKED ISSUE #3785: Upgrade Tonic to stable version + add CORS
See https://github.com/hyperium/tonic/commits/master

### PR DESCRIPTION
**Pull request by aoudiamoncef**
* [x] document all added functions
* [x] try in sandbox /simulation/labnet
* [x] unit tests on the added/changed features
  * [x] make tests compile
  * [x] make tests pass 
* [x] add logs allowing easy debugging in case the changes caused problems
* [x] if the API has changed, update the API specification

closes #3785

### CODE REVIEW

**Comment by aoudiamoncef** [comment]
@AurelienFT this PR brings CORS support + improvements in doc

**Review by AurelienFT — APPROVED** [review]

**Comment by AurelienFT** [comment]
Can you try in sandbox that it works well ? And also closing node etc

File: massa-grpc/src/server.rs [hunk 1]
```diff
@@ -75,22 +77,49 @@ impl MassaGrpc {
...
+                    .allow_origin(Any)
+                    .allow_headers([hyper::header::CONTENT_TYPE]);
+
+                let mut router_with_http1 = server_builder
+                    .accept_http1(true)
+                    .layer(cors)
+                    .layer(GrpcWebLayer::new())
+                    .add_service(svc);
+
+                if config.enable_reflection {
```
Comment (modship) [inline massa-grpc/src/server.rs:94]: maybe can refactor the build of router_with_http1 for avoid duplication

**Review by modship — APPROVED** [review]

File: massa-grpc/src/server.rs (hunk 1 above)
Comment (aoudiamoncef) [reply massa-grpc/src/server.rs:94]: I have the same issue.. builder is not a builder pattern but it's a typestate pattern. It's impossible to optimize

**Comment by aoudiamoncef** [comment]
> Can you try in sandbox that it works well ? And also closing node etc

Already done in sandbox, no regression detected

**Pull request merged**